core/Config.php

```diff
@@ -215,28 +229,42 @@ public static function getHostname()
      *     $config->save();
      *
      * @param string $hostname eg piwik.example.com
+     * @param string $preferredPath If there are different paths for the config that can be used, eg /config/* and /misc/user/*,
+     *                              and a preferred path is given, then the config path must contain the preferred path.
      * @return string
      * @throws \Exception In case the domain contains not allowed characters
      * @internal
      */
-    public function forceUsageOfLocalHostnameConfig($hostname)
+    public function forceUsageOfLocalHostnameConfig($hostname, $preferredPath = null)
     {
-        $hostConfig = self::getLocalConfigInfoForHostname($hostname);
+        $hostConfigs = self::getLocalConfigInfoForHostname($hostname);
+        $fileNames = '';
+
+        foreach ($hostConfigs as $hostConfig) {
+            if (count($hostConfigs) > 1
+                && $preferredPath
+                && strpos($hostConfig['path'], $preferredPath) === false) {
+                //
```

Comment:
Was there supposed to be a comment here?

Reply:
will remove it